### PR TITLE
Search by Category: unit tests

### PR DESCRIPTION
**Pull request by SaiShan**
User Story: #54 
Task: #114

### CODE REVIEW

**Review by OlivierTD — CHANGES_REQUESTED** [review]
Please review your tests thanks!

**Review by davidlegare — CHANGES_REQUESTED** [review]
You can't mock the fragment that you are testing, it beats the whole purpose of testing it, you just mock the dependencies.

**Comment by davidlegare** [comment]
There are conflicts to resolve as well.

File: app/src/test/java/categorySearchTest/CategorySearchTest.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        CharSequence [] ca = {"arts", "comedy","education"};
+        afFrag = mock (AddFeedFragment.class);
+        when(afFrag.getCategories()).thenReturn(ca);
+        assertEquals(afFrag.getCategories(),ca);
+
+    }
+
+    //testing the creation of an alert dialog
+    @Test
+    public void builderTest() throws Exception{
```
[the remark this answers is not in the document]
Comment (SaiShan) [reply app/src/test/java/categorySearchTest/CategorySearchTest.java]: From my research, this is the only kind of test that i have found for alert dialog. Other functions for this user story, such as the search, have the same issues as the centralized search.

**Pull request closed without merging**